src/main/host/managed_thread.rs

```diff
@@ -183,16 +183,41 @@ impl ManagedThread {
         loop {
             match *self.current_event.borrow() {
                 ShimEventToShadow::Null => {
-                    // send the message to the shim to call main().
-                    // The plugin will run until it makes a blocking call.
-                    trace!("sending start event code to {}", self.native_pid.unwrap());
-                    self.continue_plugin(
-                        ctx.host,
-                        &ShimEventToShim::Start(ShimEventStart {
-                            thread_shmem_block: ctx.thread.shmem().serialize(),
-                            process_shmem_block: ctx.process.shmem().serialize(),
-                        }),
+                    // Initialize the shim.
+
+                    // `wait_for_next_event` expects shmem to be unlocked.
+                    ctx.host.unlock_shmem();
```

Comment:
It seems a little confusing to have an `unlock` without a corresponding `lock`. Can we unlock this closer to where it was previously locked and used?